presto-main/src/main/java/com/facebook/presto/operator/DriverContext.java

```diff
@@ -159,24 +159,23 @@ public void recordBlocked(ListenableFuture<?> blocked)
 
     public void finished()
     {
+        freeMemory(memoryReservation.get());
```

Comment:
I'd like use `freeMemory(memoryReservation.getAndSet(0))` but `freeMemory` decrease the variable in it. Anyway, basic assumption here is, `finished` is called in a guard from `Driver.destroyIfNecessary` and the first call of `freeMemory` will set the variable 0. 

`DriverContext.failed` set `finished variable` true, so there still need a place to free memory regardless the state.